### PR TITLE
More array functions

### PR DESCRIPTION
**Pull request by nirvn**
## Description
This PR offers two additional array functions:
- array_sort() : used to sort an array
- ~~array_unique_values() : used to output an array containing all unique values of a provided array (i.e., duplicate values removed).~~ so turns out I implemented this a while ago under the function name array_distinct and I had totally forgotten about it. Checking in a clinic in the coming weeks ;)

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by DelazJ** [comment]
@nirvn what about a second parameter to sort desc?

**Comment by nirvn** [comment]
@DelazJ , yeah, thought about it, but thought it might be better to have ppl use de preexisting array_reverse function.

**Comment by Gustry** [comment]
@nirvn What about your previous work? https://github.com/qgis/QGIS/commit/17fc7dbe53367f66c0b68930b6d278e41d896b49#diff-09242f2401f3c0beb48deac90241f26d ;-)

**Comment by nirvn** [comment]
Wh-what? God I'm going senile at the age of 38!

**Comment by nirvn** [comment]
@DelazJ , here, request accepted:
image

File: src/core/expression/qgsexpressionfunction.cpp [hunk 1]
```diff
@@ -4104,6 +4104,14 @@ static QVariant fcnArray( const QVariantList &values, const QgsExpressionContext
   return values;
 }
 
+static QVariant fcnArraySort( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QVariantList list = QgsExpressionUtils::getListValue( values.at( 0 ), parent );
+  bool reverted = values.value( 1 ).toBool();
```
Comment (nyalldawson) [inline src/core/expression/qgsexpressionfunction.cpp]: "reversed"

**Review by nyalldawson — COMMENTED** [review]
Looks good!

**Comment by DelazJ** [comment]
I wonder if the logic should not be the reverse. I mean, writing `array_sort( array( 10, 50, 6 ), true) ` i'd expect the result to be sorted from down to top ie `[6,10,50]` and not `[50,10,6]`. Getting a descending order with associating "sort" and "true" does not look _obvious_ (or _natural_?)

**Comment by nirvn** [comment]
@DelazJ , frankly, `array_sort( array( 10, 50, 6 ), true/false )` doesn't leave me with any expectations / impressions. Getting a descending order with associating "sort" and "false" doesn't look any more obvious or natural to me :) That's why I'd suggested to use named argument if you ever write documentation or blog post on this, i.e. `array_sort( array( 10, 50, 6 ), reversed:=true )`.

File: resources/function_help/json/array_sort [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "array_sort",
+  "type": "function",
+  "description": "Returns the provided array with its elements sorted.",
+  "arguments": [ {"arg":"array","description":"an array"},
+                 {"arg":"reversed","optional":true,"default":"false","description":"set this parameter to true to sort the array in reverse order"}],
```
[the remark this answers is not in the document]
Comment (nirvn) [reply resources/function_help/json/array_sort]: @DelazJ , good point, updated.

**Comment by Gustry** [comment]
I would agree with @DelazJ and I think it would be nicer to use the same convention as existing expression/algorithm. In the sort by expression algorithm, the keyword is "ascending" default to true.

https://github.com/qgis/QGIS/blob/master/src/analysis/processing/qgsalgorithmorderbyexpression.cpp#L54

**Pull request merged**